images/temporal-server/config/main.tf

```diff
@@ -0,0 +1,26 @@
+terraform {
+  required_providers {
+    apko = { source = "chainguard-dev/apko" }
+  }
+}
+
+variable "extra_packages" {
+  description = "The additional packages to install"
+  default = [
+    "temporal-server",
+  ]
+}
+
+module "accts" { source = "../../../tflib/accts" }
+
+output "config" {
+  value = jsonencode({
+    contents = {
+      packages = concat(var.extra_packages)
+    }
+    accounts = module.accts.block
+    entrypoint = {
+      command = "/usr/bin/temporal-server"
```

Comment:
The upstream entrypoint has a lot of interesting config templating stuff going on: https://oci.dag.dev/layers/temporalio/server@sha256:db23904aab3d6b295f7971e71bd276a156b5b1c0ea02f9d547ff8bbde331dc37/etc/temporal/entrypoint.sh
https://oci.dag.dev/layers/temporalio/server@sha256:db23904aab3d6b295f7971e71bd276a156b5b1c0ea02f9d547ff8bbde331dc37/etc/temporal/config/config_template.yaml

I suspect some of this might be required for it to work as expected?

Reply:
Yes, initially I thought users will provide the config using the flag but I guess you are right, where should we add this? I am not sure if we can do this here in image part rather it should be part of the package and corrected there? Have we faced or done anything similar before @jonjohnsonjr ? That would be a great help!